src/webfaf/user.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2018  ABRT Team
+# Copyright (C) 2018  Red Hat, Inc.
+#
+# This file is part of faf.
+#
+# faf is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# faf is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with faf.  If not, see <http://www.gnu.org/licenses/>.
+
+from flask import abort, Blueprint, flash, g, make_response, redirect, session, url_for
+from pyfaf import queries
+from pyfaf.utils.user import UserDataDumper
+from webfaf.webfaf_main import db, oid
+from webfaf.utils import (delete_complete_report,
+                          delete_user_bugzillas,
+                          login_required)
+
+user = Blueprint("user", __name__)
+
+
+@user.route("/delete_data", endpoint='delete')
+@login_required
+def delete_user_data():
+    usermail = g.user.mail
+    fas_user = queries.get_user_by_mail(db, usermail).first()
+    bz_user = queries.get_bz_user(db, usermail)
+    contact_email = queries.get_contact_email(db, usermail)
+
+    if bz_user is not None:
+        delete_user_bugzillas(db, bz_user.id)
+        db.session.delete(bz_user)
+
+    if fas_user is not None:
+        queries.get_reportarchives_by_username(db, fas_user.username).delete(False)
+        queries.get_problemreassigns_by_username(db, fas_user.username).delete(False)
+
+    if contact_email is not None:
+        user_reports = queries.get_reportcontactmails_by_id(db, contact_email.id)
+
+        for report in user_reports.all():
+            delete_complete_report(db, report.report_id)
```

Comment:
soo, you remove the report, if the user attached his/hers email to it? This is soo unsafe, because I can attach my email to any existing report (for example look at `reporter-ureport --help` how easy is to do that) and then remove any report. This way I can clear our db easily :)

Reply:
Now I saw, that in the given function there is the check if the count is more than one, little bit confusing to call method `delete_complete_report` but delete only when the count is less than 2 :) Still there is the problem I can remove selected report, but sure, only if the count is 1.

Reply:
If the `count` is 1 and there is a `contact mail` linked to it then the reports comes from the user, right?



> little bit confusing to call method delete_complete_report but delete only when the count is less than 2 :)

I will change that.

Reply:
not necessary. Under normal circumstances it is true, but anyone can attach its email to any report. So I can pick any report on faf with count==1, use reporter-ureport to attach my email to it and then remove it.